### ISSUE TITLE
type_annotate_public_apis reports issues over local non-API functions

### ISSUE THREAD

**Issue by chalin** [dart-lang/linter#1108]
Consider the local `good()` function declared in this test:

```dart
test('foo', () {
  good() { ... }
  expect(good, ...);
});
```

The `type_annotate_public_apis` lint causes this issue to be reported:

```nocode
lint • Type annotate public APIs at test/effective_dart_test.dart:28:7 • type_annotate_public_apis
```

Being a local function, `good()` should not be considered in this lint.

cc @munificent @kevmoo

**Comment by kevmoo** [comment]
Agreed!

**Comment by pq** [comment]
+1.

Good catch!

**Comment by chalin** [comment]
Can you add a comment here once this is released publicly?

**Comment by srawlins** [comment]
I can't promise that. I wish there was a bot for such things.

**Comment by pq** [comment]
@chalin, @srawlins: I'm curious if you have any thoughts on how we could do better at documenting and publicizing releases.  I'm very open to a bot actually if we could decide what exactly it should do.

One thing that occurs to me right away is that the published lint docs reflect HEAD but there's no easy way to see what is defined in a particular SDK.  I don't want to clutter the docs w/ badges but maybe something could be done?

![image](https://user-images.githubusercontent.com/67586/50654879-289b4280-0f43-11e9-8871-cddcd7578866.png)


For example, what if we added an SDK badge w/ the number of the minimum supporting SDK?

**Comment by kevmoo** [comment]
> what if we added an SDK badge w/ the number of the minimum supporting SDK?

❤️ ❤️ ❤️

**Comment by srawlins** [comment]
+10, love that idea. Would be very simple to implement with a String API on each rule.
